<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Getting Started with KDPS</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Getting Started with KDPS</h1>



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p><code>KDPS</code> (Kinship Decouple and Phenotype Selection) is an R
package that resolves cryptic relatedness in genetic studies using a
phenotype-aware approach. It removes related individuals based on
kinship or IBD scores while prioritizing the retention of subjects with
phenotypes of interest.</p>
<p>This tool is useful in GWAS and epidemiological studies where
maximizing the number of unrelated individuals with relevant traits is
essential for statistical power, especially in rare or stratified
phenotypes.</p>
<hr />
</div>
<div id="installation" class="section level1">
<h1>Installation</h1>
<p>To install the latest version from GitHub:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;devtools&quot;</span>)){</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>  <span class="fu">library</span>(<span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>}</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="co">#&gt; Loading required package: devtools</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="co">#&gt; Loading required package: usethis</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;kdps&quot;</span>)){</span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;UCSD-Salem-Lab/kdps&quot;</span>)</span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a>  <span class="fu">library</span>(<span class="st">&quot;kdps&quot;</span>)</span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a>}</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="co">#&gt; Loading required package: kdps</span></span></code></pre></div>
<hr />
</div>
<div id="example-data" class="section level1">
<h1>Example Data</h1>
<p>This package includes two example files in <code>extdata/</code>:</p>
<div id="simple_pheno.txt" class="section level3">
<h3><code>simple_pheno.txt</code></h3>
<p>This file contains phenotypic data for individuals in the cohort.
Each row represents one individual.</p>
<table>
<thead>
<tr class="header">
<th>Column</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>FID</code></td>
<td>Family ID (used for linking with kinship data)</td>
</tr>
<tr class="even">
<td><code>IID</code></td>
<td>Individual ID</td>
</tr>
<tr class="odd">
<td><code>pheno1</code></td>
<td>A binary phenotype (e.g., disease status)</td>
</tr>
<tr class="even">
<td><code>pheno2</code></td>
<td>A categorical phenotype used in prioritization</td>
</tr>
<tr class="odd">
<td><code>pheno3</code></td>
<td>A continuous trait (e.g., height or biomarker)</td>
</tr>
</tbody>
</table>
<p>Example:</p>
<table>
<thead>
<tr class="header">
<th>FID</th>
<th>IID</th>
<th>pheno1</th>
<th>pheno2</th>
<th>pheno3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>0</td>
<td>1001</td>
<td>DISEASED</td>
<td>DISEASED2</td>
<td>109.5</td>
</tr>
<tr class="even">
<td>0</td>
<td>1002</td>
<td>HEALTHY</td>
<td>HEALTHY</td>
<td>117.18</td>
</tr>
<tr class="odd">
<td>0</td>
<td>1003</td>
<td>HEALTHY</td>
<td>HEALTHY</td>
<td>90.41</td>
</tr>
<tr class="even">
<td>0</td>
<td>1004</td>
<td>HEALTHY</td>
<td>HEALTHY</td>
<td>95</td>
</tr>
</tbody>
</table>
<hr />
</div>
<div id="simple_kinship.txt" class="section level3">
<h3><code>simple_kinship.txt</code></h3>
<p>This file encodes pairwise relatedness between individuals based on
genome-wide genotype data.</p>
<table>
<colgroup>
<col width="45%" />
<col width="54%" />
</colgroup>
<thead>
<tr class="header">
<th>Column</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>FID1</code></td>
<td>Family ID of individual 1</td>
</tr>
<tr class="even">
<td><code>IID1</code></td>
<td>Individual ID of individual 1</td>
</tr>
<tr class="odd">
<td><code>FID2</code></td>
<td>Family ID of individual 2</td>
</tr>
<tr class="even">
<td><code>IID2</code></td>
<td>Individual ID of individual 2</td>
</tr>
<tr class="odd">
<td><code>HetHet</code></td>
<td>Proportion of sites where both individuals are heterozygous</td>
</tr>
<tr class="even">
<td><code>IBS0</code></td>
<td>Proportion of sites with no alleles in common</td>
</tr>
<tr class="odd">
<td><code>KINSHIP</code></td>
<td>Estimated kinship coefficient (values &gt; 0.0442 typically indicate
2nd-degree or closer relationships)</td>
</tr>
</tbody>
</table>
<p>Example:</p>
<table>
<thead>
<tr class="header">
<th>FID1</th>
<th>IID1</th>
<th>FID2</th>
<th>IID2</th>
<th>HetHet</th>
<th>IBS0</th>
<th>KINSHIP</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>0</td>
<td>1001</td>
<td>0</td>
<td>1002</td>
<td>0.037</td>
<td>0.0083</td>
<td>1</td>
</tr>
<tr class="even">
<td>0</td>
<td>1003</td>
<td>0</td>
<td>1004</td>
<td>0.046</td>
<td>0.0148</td>
<td>1</td>
</tr>
</tbody>
</table>
<hr />
</div>
</div>
<div id="simple-example-resolving-relatedness-in-a-small-cohort" class="section level1">
<h1>Simple Example: Resolving Relatedness in a Small Cohort</h1>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">library</span>(kdps)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>phenotype_file <span class="ot">=</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;simple_pheno.txt&quot;</span>, <span class="at">package =</span> <span class="st">&quot;kdps&quot;</span>)</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>kinship_file   <span class="ot">=</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;simple_kinship.txt&quot;</span>, <span class="at">package =</span> <span class="st">&quot;kdps&quot;</span>)</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>kdps_results <span class="ot">=</span> <span class="fu">kdps</span>(</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>  <span class="at">phenotype_file =</span> phenotype_file,</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a>  <span class="at">kinship_file =</span> kinship_file,</span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>  <span class="at">fuzziness =</span> <span class="dv">0</span>,</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>  <span class="at">phenotype_name =</span> <span class="st">&quot;pheno2&quot;</span>,</span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>  <span class="at">prioritize_high =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>  <span class="at">prioritize_low =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a>  <span class="at">phenotype_rank =</span> <span class="fu">c</span>(<span class="st">&quot;DISEASED1&quot;</span>, <span class="st">&quot;DISEASED2&quot;</span>, <span class="st">&quot;HEALTHY&quot;</span>),</span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a>  <span class="at">fid_name =</span> <span class="st">&quot;FID&quot;</span>,</span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a>  <span class="at">iid_name =</span> <span class="st">&quot;IID&quot;</span>,</span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a>  <span class="at">fid1_name =</span> <span class="st">&quot;FID1&quot;</span>,</span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a>  <span class="at">iid1_name =</span> <span class="st">&quot;IID1&quot;</span>,</span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a>  <span class="at">fid2_name =</span> <span class="st">&quot;FID2&quot;</span>,</span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a>  <span class="at">iid2_name =</span> <span class="st">&quot;IID2&quot;</span>,</span>
<span id="cb2-20"><a href="#cb2-20" tabindex="-1"></a>  <span class="at">kinship_name =</span> <span class="st">&quot;KINSHIP&quot;</span>,</span>
<span id="cb2-21"><a href="#cb2-21" tabindex="-1"></a>  <span class="at">kinship_threshold =</span> <span class="fl">0.0442</span>,</span>
<span id="cb2-22"><a href="#cb2-22" tabindex="-1"></a>  <span class="at">phenotypic_naive =</span> <span class="cn">FALSE</span></span>
<span id="cb2-23"><a href="#cb2-23" tabindex="-1"></a>)</span>
<span id="cb2-24"><a href="#cb2-24" tabindex="-1"></a></span>
<span id="cb2-25"><a href="#cb2-25" tabindex="-1"></a>kdps_results</span></code></pre></div>
<hr />
</div>
<div id="complex-example-larger-relatedness-network" class="section level1">
<h1>Complex Example: Larger Relatedness Network</h1>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>phenotype_file <span class="ot">=</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;complex_pheno.txt&quot;</span>, <span class="at">package =</span> <span class="st">&quot;kdps&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>kinship_file   <span class="ot">=</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;complex_kinship.txt&quot;</span>, <span class="at">package =</span> <span class="st">&quot;kdps&quot;</span>)</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>kdps_results <span class="ot">=</span> <span class="fu">kdps</span>(</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>  <span class="at">phenotype_file =</span> phenotype_file,</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>  <span class="at">kinship_file =</span> kinship_file,</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>  <span class="at">fuzziness =</span> <span class="dv">0</span>,</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>  <span class="at">phenotype_name =</span> <span class="st">&quot;pheno2&quot;</span>,</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>  <span class="at">prioritize_high =</span> <span class="cn">FALSE</span>,</span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>  <span class="at">prioritize_low =</span> <span class="cn">FALSE</span>,</span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>  <span class="at">phenotype_rank =</span> <span class="fu">c</span>(<span class="st">&quot;DISEASED1&quot;</span>, <span class="st">&quot;DISEASED2&quot;</span>, <span class="st">&quot;HEALTHY&quot;</span>),</span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>  <span class="at">fid_name =</span> <span class="st">&quot;FID&quot;</span>,</span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>  <span class="at">iid_name =</span> <span class="st">&quot;IID&quot;</span>,</span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>  <span class="at">fid1_name =</span> <span class="st">&quot;FID1&quot;</span>,</span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a>  <span class="at">iid1_name =</span> <span class="st">&quot;IID1&quot;</span>,</span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a>  <span class="at">fid2_name =</span> <span class="st">&quot;FID2&quot;</span>,</span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a>  <span class="at">iid2_name =</span> <span class="st">&quot;IID2&quot;</span>,</span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a>  <span class="at">kinship_name =</span> <span class="st">&quot;KINSHIP&quot;</span>,</span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a>  <span class="at">kinship_threshold =</span> <span class="fl">0.0442</span>,</span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a>  <span class="at">phenotypic_naive =</span> <span class="cn">FALSE</span></span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a>)</span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a></span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a><span class="fu">head</span>(kdps_results)</span></code></pre></div>
<hr />
</div>
<div id="function-arguments" class="section level1">
<h1>Function Arguments</h1>
<p>Key arguments for <code>kdps()</code> include:</p>
<ul>
<li><code>phenotype_file</code>, <code>kinship_file</code>: File paths
to phenotype and kinship matrices.</li>
<li><code>phenotype_name</code>: The column name of the phenotype to
prioritize.</li>
<li><code>phenotype_rank</code>: Ordered levels from most to least
important.</li>
<li><code>kinship_threshold</code>: Kinship score above which subjects
are considered related.</li>
<li><code>fuzziness</code>: Controls tolerance when resolving complex
networks (default = 0).</li>
<li><code>prioritize_high</code>, <code>prioritize_low</code>: If
<code>TRUE</code>, prioritizes subjects with extreme phenotype values
(numeric).</li>
<li><code>phenotypic_naive</code>: If <code>TRUE</code>, phenotype info
is ignored and ties are broken randomly.</li>
</ul>
<hr />
</div>
<div id="output" class="section level1">
<h1>Output</h1>
<p>The output is a <code>data.frame</code> with columns:</p>
<ul>
<li><code>FID</code>: Family ID of the subject to remove.</li>
<li><code>IID</code>: Individual ID of the subject to remove.</li>
</ul>
<p>You can save this output to a text file to filter out individuals in
your downstream analysis.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">write.table</span>(kdps_results, <span class="at">file =</span> <span class="st">&quot;subjects_to_remove.txt&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<hr />
</div>
<div id="final-notes" class="section level1">
<h1>Final Notes</h1>
<ul>
<li>KDPS is designed for large-scale studies like UK Biobank, with
efficient performance even on complex networks.</li>
<li>Users are encouraged to interpret results in the context of
potential <strong>collider bias</strong> introduced by phenotype-aware
filtering.</li>
<li>For large studies, consider pre-filtering unrelated individuals
using tools like <code>PLINK</code> and using KDPS for final
refinement.</li>
</ul>
<hr />
<p>For updates and source code, visit: <a href="https://github.com/UCSD-Salem-Lab/kdps%22">https://github.com/UCSD-Salem-Lab/kdps</a></p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
